dali/pipeline/operators/python_function/dltensor_function.h

```diff
@@ -106,13 +106,17 @@ class DLTensorPythonFunctionImpl : public PythonFunctionImplBase<Backend> {
     py::gil_scoped_acquire interpreter_guard{};
     py::object output_o;
     try {
-      output_o = python_function(*py::tuple(detail::PrepareDLTensorInputs<Backend>(ws)));
+      if (std::is_same<Backend, GPUBackend>::value) {
+        cudaStreamSynchronize(ws.stream());
+        SetCurrentStream(ws.stream());
+      }
+      output_o = python_function(*detail::PrepareDLTensorInputs<Backend>(ws));
```

Comment:
I would revert to the old stream here. You can use some RAII object to do that.

Reply:
done